internal/command/group_member.go

```diff
@@ -0,0 +1,225 @@
+package command
+
+import (
+	"context"
+
+	"github.com/zitadel/zitadel/internal/domain"
+	"github.com/zitadel/zitadel/internal/eventstore"
+	"github.com/zitadel/zitadel/internal/repository/group"
+	"github.com/zitadel/zitadel/internal/repository/user"
+	"github.com/zitadel/zitadel/internal/telemetry/tracing"
+	"github.com/zitadel/zitadel/internal/zerrors"
+)
+
+func (c *Commands) AddGroupMember(ctx context.Context, member *domain.Member, resourceOwner string) (_ *domain.Member, err error) {
+	ctx, span := tracing.NewSpan(ctx)
+	defer func() { span.EndWithError(err) }()
+
+	addedMember := NewGroupMemberWriteModel(member.AggregateID, member.UserID, resourceOwner)
+	addedGroupMember := NewUserGroupMemberWriteModel(member.AggregateID, member.UserID, resourceOwner)
+
+	groupAgg := GroupAggregateFromWriteModel(&addedMember.WriteModel)
+	userGroupAgg := GroupAggregateFromWriteModel(&addedGroupMember.WriteModel)
+
+	event, err := c.addGroupMember(ctx, groupAgg, addedMember, member)
+	if err != nil {
+		return nil, err
+	}
+
+	userEvent, err := c.addUserGroupMember(ctx, userGroupAgg, addedGroupMember, member.AggregateID)
```

Comment:
Why are there 2 different types of member events? Do these try to achieve the same thing?